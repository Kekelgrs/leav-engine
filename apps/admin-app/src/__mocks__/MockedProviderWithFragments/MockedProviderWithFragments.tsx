// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {MockedProvider} from '@apollo/react-testing';
import {InMemoryCache} from 'apollo-cache-inmemory';
import React from 'react';
import {attributesFragmentMatcher} from '../fragmentMatchers/attributesFragmentMatchers';

export const getMockCacheWithFragments = () => new InMemoryCache({fragmentMatcher: attributesFragmentMatcher});

function MockedProviderWithFragments({children, ...props}) {
    // Set a new cache for each test to avoid fetching data in cache and not in provided mocks
    const mockCache = getMockCacheWithFragments();
    return (
        <MockedProvider cache={mockCache} {...props}>
            {children}
        </MockedProvider>
    );
}

export default MockedProviderWithFragments;
