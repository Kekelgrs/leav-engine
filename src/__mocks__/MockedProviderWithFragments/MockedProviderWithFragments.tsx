// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCache} from '@apollo/client';
import {MockedProvider} from '@apollo/client/testing';
import React from 'react';

// export const getMockCacheWithFragments = () => new InMemoryCache({fragmentMatcher: attributesFragmentMatcher});
export const getMockCacheWithFragments = () => new InMemoryCache();

function MockedProviderWithFragments({children, ...props}: any) {
    // Set a new cache for each test to avoid fetching data in cache and not in provided mocks
    const mockCache = getMockCacheWithFragments();
    return (
        <MockedProvider cache={mockCache} {...props}>
            {children}
        </MockedProvider>
    );
}

export default MockedProviderWithFragments;
