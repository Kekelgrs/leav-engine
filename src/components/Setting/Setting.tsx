import {useQuery} from '@apollo/client';
import React from 'react';
import {useTranslation} from 'react-i18next';
import {Container, Dropdown, DropdownProps, Header, Segment} from 'semantic-ui-react';
import {getAvailableLangs, getLangAndDefaultLang} from '../../queries/cache/lang/getLangQuery';
import {AvailableLanguage} from '../../_types/types';

function Setting(): JSX.Element {
    const {t, i18n: i18nClient} = useTranslation();

    const {data: dataLang, client} = useQuery(getAvailableLangs);
    const {availableLangs, lang} = dataLang ?? {availableLangs: [], lang: []};

    const langOption = availableLangs.map((l: string) => ({
        key: l,
        value: l,
        text: l
    }));

    const changeLang = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
        i18nClient.changeLanguage(data.value ?? (lang[0] as any));

        // Update cache lang infos
        const newLang = [i18nClient.language, i18nClient.language];

        const defaultLang = i18nClient.language
            ? AvailableLanguage[i18nClient.language as AvailableLanguage]
            : AvailableLanguage.en;

        client.writeQuery({
            query: getLangAndDefaultLang,
            data: {
                lang: newLang,
                defaultLang
            }
        });
    };

    return (
        <Container>
            <Header>{t('settings.header')}</Header>

            <Segment>
                <Dropdown selection fluid options={langOption} defaultValue={lang[0]} onChange={changeLang} />
            </Segment>
        </Container>
    );
}

export default Setting;
