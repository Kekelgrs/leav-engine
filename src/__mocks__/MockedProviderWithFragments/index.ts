import MockedProviderWithFragments from './MockedProviderWithFragments';
export default MockedProviderWithFragments;
