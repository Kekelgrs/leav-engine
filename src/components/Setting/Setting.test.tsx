import {render} from 'enzyme';
import React from 'react';
import MockedProviderWithFragments from '../../__mocks__/MockedProviderWithFragments';
import Setting from './Setting';

describe('Setting', () => {
    test('Snapshot test', async () => {
        const comp = render(
            <MockedProviderWithFragments>
                <Setting />
            </MockedProviderWithFragments>
        );

        expect(comp).toMatchSnapshot();
    });
});
