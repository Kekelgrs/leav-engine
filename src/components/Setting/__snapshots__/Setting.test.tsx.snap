// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Snapshot test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ui container",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ui header",
        },
        "children": Array [
          Object {
            "data": "settings.header",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ui segment",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-expanded": "false",
                "class": "ui fluid selection dropdown",
                "role": "listbox",
                "tabindex": "0",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-atomic": "true",
                    "aria-live": "polite",
                    "class": "text",
                    "role": "alert",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "dropdown icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "class": "menu transition",
                      },
                      "children": Array [],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-atomic": undefined,
                    "aria-live": undefined,
                    "class": undefined,
                    "role": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-atomic": undefined,
                    "aria-live": undefined,
                    "class": undefined,
                    "role": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "dropdown icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "menu transition",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-atomic": "true",
                      "aria-live": "polite",
                      "class": "text",
                      "role": "alert",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-atomic": undefined,
                      "aria-live": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-atomic": undefined,
                      "aria-live": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "menu transition",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "dropdown icon",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "aria-atomic": "true",
                        "aria-live": "polite",
                        "class": "text",
                        "role": "alert",
                      },
                      "children": Array [],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-atomic": undefined,
                        "aria-live": undefined,
                        "class": undefined,
                        "role": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-atomic": undefined,
                        "aria-live": undefined,
                        "class": undefined,
                        "role": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-expanded": undefined,
                "class": undefined,
                "role": undefined,
                "tabindex": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-expanded": undefined,
                "class": undefined,
                "role": undefined,
                "tabindex": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ui segment",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-expanded": "false",
              "class": "ui fluid selection dropdown",
              "role": "listbox",
              "tabindex": "0",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-atomic": "true",
                  "aria-live": "polite",
                  "class": "text",
                  "role": "alert",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "dropdown icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "menu transition",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-atomic": undefined,
                  "aria-live": undefined,
                  "class": undefined,
                  "role": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-atomic": undefined,
                  "aria-live": undefined,
                  "class": undefined,
                  "role": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "dropdown icon",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "menu transition",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-atomic": "true",
                    "aria-live": "polite",
                    "class": "text",
                    "role": "alert",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-atomic": undefined,
                    "aria-live": undefined,
                    "class": undefined,
                    "role": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-atomic": undefined,
                    "aria-live": undefined,
                    "class": undefined,
                    "role": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "menu transition",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "dropdown icon",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "aria-atomic": "true",
                      "aria-live": "polite",
                      "class": "text",
                      "role": "alert",
                    },
                    "children": Array [],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-atomic": undefined,
                      "aria-live": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-atomic": undefined,
                      "aria-live": undefined,
                      "class": undefined,
                      "role": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-expanded": undefined,
              "class": undefined,
              "role": undefined,
              "tabindex": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-expanded": undefined,
              "class": undefined,
              "role": undefined,
              "tabindex": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ui header",
          },
          "children": Array [
            Object {
              "data": "settings.header",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
